junit5/src/main/java/org/jboss/weld/junit5/auto/ClassScanning.java

```diff
@@ -199,14 +199,10 @@ static void scanForRequiredBeanClass(Class<?> testClass, Weld weld, boolean expl
 
         }
 
-        foundClasses.add(testClass);
```

Comment:
Hmm, I wonder why we had this here before. In theory you would get the test class processed either way as part of `classesToProcess`...

Reply:
I think that this line was superfluous already before this MR because test classes were added once here to weld when `hasBeanDefiningAnnotation` further below

https://github.com/philippkunz/weld-junit/blob/nested-readme/junit5/src/main/java/org/jboss/weld/junit5/auto/ClassScanning.java#L203-205 vs. https://github.com/weld/weld-junit/blob/master/junit5/src/main/java/org/jboss/weld/junit5/auto/ClassScanning.java#L206-L208

            if (hasBeanDefiningAnnotation(foundClass)) {
                weld.addBeanClass(foundClass);
            }

and once more in

https://github.com/philippkunz/weld-junit/blob/nested-readme/junit5/src/main/java/org/jboss/weld/junit5/auto/WeldJunit5AutoExtension.java#L61

        weld.addBeanClasses(testClasses.stream().toArray(Class<?>[]::new));

vs. https://github.com/weld/weld-junit/blob/master/junit5/src/main/java/org/jboss/weld/junit5/auto/WeldJunit5AutoExtension.java#L57

        weld.addBeanClass(testClass);

the second time without an additional condition. I'm pretty confident I haven't broken anything considering test classes are scanned for other used beans classes with 

https://github.com/weld/weld-junit/blob/master/junit5/src/main/java/org/jboss/weld/junit5/auto/ClassScanning.java#L70

        classesToProcess.add(testClass);

but which does not make them beans to be discovered for auto deployment necessarily.

I think it was slightly odd before and hope this does not point to a more serious issue. I also wonder why the test classes should be added in `WeldJunit5AutoExtension` explicitly or if they should rather be detected during `ClassScanning.scanForRequiredBeanClass` but that might be a different topic.

Reply:
As I think about it again, it occurs to me after you pointed it out that I might have removed the correct code rather than the wrong one. Shall I change it back?

Reply:
Upon closer inspection, I think this line can be removed because a test class will:
a) Always be added and inspected in the first few steps
b) Will never pass the `hasBeanDefiningAnnotation(foundClass)` condition anyway (which is intended!)

Reply:
* to a): assuming b), a) has no effect because of `if (hasBeanDefiningAnnotation(foundClass)) {` later.
* to b): I don't agree that's a correct assumption. A test class could come with anything making it pass that condition. It only might be unlikely that many of them currently exist with a scope that conflicts with `@Singleton`. Consider the following example:

```java
package org.jboss.weld.junit5.auto.classscanning;

import static java.lang.annotation.ElementType.TYPE;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import javax.enterprise.inject.Stereotype;

import org.jboss.weld.junit5.auto.EnableAutoWeld;
import org.junit.jupiter.api.Test;

@EnableAutoWeld
@TestClassStereotypeTest.CustomStereotype
public class TestClassStereotypeTest {

    @Stereotype
    @Target(TYPE)
    @Retention(RUNTIME)
    public @interface CustomStereotype {
    }
    

    @Test
    void test() {
        
    }

}
```

For whatever reason, I agree that `foundClasses.add(testClass);` should be removed.